### PR TITLE
BaSP-MR-109: member and activities react-hook-form

### PR DESCRIPTION
**Pull request by Dannylez**
Member: new forms and joi validations.
Activity: new forms and joi validations.

### CODE REVIEW

File: src/Components/Members/Edit/EditMembers.jsx [hunk 1]
```diff
@@ -49,20 +63,82 @@ const MembersEdit = () => {
...
+        'string.pattern.base': 'Phone number must contain only numbers'
+      }),
+    email: Joi.string()
+      .max(30)
+      .pattern(/^[^@]+@[^@]+\.[a-zA-Z]{2,}$/)
+      .messages({
+        'string.pattern.base': 'Invalid email format'
+      }),
+    city: Joi.string().min(2).max(30),
+    birthDay: Joi.date().min('1930-01-01').max('2008-01-01'),
```
Comment (DelfiLT) [inline src/Components/Members/Edit/EditMembers.jsx:102]: The birth validation isn't working well, the input doesn't says the error and instead I received a modal 
image

Comment (DelfiLT) [reply src/Components/Members/Edit/EditMembers.jsx:102]: Also, if you are using the joi form errors, it shouldn't be a modal

File: src/Components/Activities/Form/index.jsx [hunk 2]
```diff
@@ -46,59 +78,57 @@ const Form = () => {
...
+      await postActivity(dispatch, data);
     }
 
     if (location.pathname.includes('edit')) {
-      await putActivity(dispatch, id, { name, description, isActive });
+      await putActivity(dispatch, id, data);
     }
   };
-
+  console.log(isActive);
```
Comment (DelfiLT) [inline src/Components/Activities/Form/index.jsx:94]: Remove console.log

File: src/Components/Members/Create/index.jsx [hunk 3]
```diff
@@ -60,20 +121,9 @@ const MembersCreate = () => {
...
-
-  const handleBirthdayChange = (e) => {
-    setMember({ ...member, birthDay: e });
-  };
-
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    dispatch(createMember(member));
+  const onSubmit = (data) => {
+    console.log(data);
```
Comment (DelfiLT) [inline src/Components/Members/Create/index.jsx:125]: Remove console.log

File: src/Components/Members/Create/index.jsx [hunk 4]
```diff
@@ -1,25 +1,64 @@
...
+        'string.pattern.base': 'Phone number must contain only numbers'
+      }),
+    email: Joi.string()
+      .max(30)
+      .pattern(/^[^@]+@[^@]+\.[a-zA-Z]{2,}$/)
+      .messages({
+        'string.pattern.base': 'Invalid email format'
+      }),
+    city: Joi.string().min(2).max(30),
+    birthDay: Joi.date().min('1930-01-01').max('2008-01-01'),
```
Comment (DelfiLT) [inline src/Components/Members/Create/index.jsx:51]: I assume the same happens here

**Review by DelfiLT — APPROVED** [review]

**Comment by ukiweskamp** [comment]
When you edit a member or activity and don't make any changes you should go back without fetching to improve performance!

**Review by Milicerro — APPROVED** [review]

**Pull request merged**